samples/kotlin-mcp-client/src/main/kotlin/io/modelcontextprotocol/sample/client/MCPClient.kt

```diff
@@ -104,11 +155,76 @@ class MCPClient : AutoCloseable {
                 .build()
         )
 
+//        val assistantReply = response.content().firstOrNull()?.text()?.getOrNull()?.text()
+
+
+//        val finalText = mutableListOf<String>()
+//        response.content().forEach { content ->
+//            when {
+//                // Append text outputs from the response
+//                content.isText() -> finalText.add(content.text().getOrNull()?.text() ?: "")
+//
+//                // If the response indicates a tool use, process it further
+//                content.isToolUse() -> {
+//                    val toolName = content.toolUse().get().name()
+//                    val toolArgs =
+//                        content.toolUse().get()._input().convert(object : TypeReference<Map<String, JsonValue>>() {})
+//
+//                    // Call the tool with provided arguments
+//                    val result = mcp.callTool(
+//                        name = toolName,
+//                        arguments = toolArgs ?: emptyMap()
+//                    )
+//                    finalText.add("[Calling tool $toolName with args $toolArgs]")
+//
+//                    // Add the tool result message to the conversation
+//                    messages.add(
+//                        MessageParam.builder()
+//                            .role(MessageParam.Role.USER)
+//                            .content(
+//                                """
+//                                        "type": "tool_result",
+//                                        "tool_name": $toolName,
+//                                        "result": ${result?.content?.joinToString("\n") { (it as TextContent).text ?: "" }}
+//                                    """.trimIndent()
+//                            )
+//                            .build()
+//                    )
+//
+//                    // Retrieve an updated response after tool execution
+//                    val aiResponse = anthropic.messages().create(
+//                        messageParamsBuilder
+//                            .messages(messages)
+//                            .build()
+//                    )
+//
+//                    // Append the updated response to final text
+//                    finalText.add(aiResponse.content().first().text().getOrNull()?.text() ?: "")
+//                }
+//            }
+//        }
+//
+//        return finalText.joinToString("\n", prefix = "", postfix = "")
+//    }
```

Comment:
Please leave a comment in the code explaining what this part does and why it is commented out